lib/firenest/topology/erlang.ex

```diff
@@ -133,112 +98,341 @@ defmodule Firenest.Topology.Erlang do
     :ets.lookup_element(topology, :nodes, 2)
   end
 
-  ## GenServer callbacks
+  # Subscribing to the topology events is private right now,
+  # we can make it public if necessary but sync_named/3 should
+  # be enough for all purposes.
+  defp subscribe(topology, pid) when is_pid(pid) do
+    GenServer.call(topology, {:subscribe, pid})
+  end
+end
+
+defmodule Firenest.Topology.Erlang.Supervisor do
+  @moduledoc false
+
+  use Supervisor
+
+  def start_link(opts) do
+    topology = Keyword.fetch!(opts, :name)
+    name = Module.concat(topology, "Supervisor")
+    Supervisor.start_link(__MODULE__, topology, name: name)
+  end
+
+  def init(topology) do
+    ^topology = :ets.new(topology, [:set, :public, :named_table, read_concurrency: true])
+    true = :ets.insert(topology, {:adapter, Firenest.Topology.Erlang})
+    children = [{Firenest.Topology.Erlang.Server, topology}]
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+end
```

Comment:
I'm not sure, I understand why there's the additional supervisor with just one child.

Reply:
Ohhhhhhh, it is because previously we said the topology would always be a supervisor because we used the `supervisor/2` but now we have child_spec, so we don't need to do this anymore. Great catch!